src/metabase/task/sync_databases.clj

```diff
@@ -254,37 +256,30 @@
   (task/add-job! sync-analyze-job)
   (task/add-job! field-values-job))
 
-(defn randomized-schedules
-  "Updated default schedules for the sync task when given the original default schedules.
-  The defaults used to be all at the same time leading to poor resource management. If the user has not indicated they
-  want to control the scheduling, returns appropriate randomized schedules for the sync tasks."
-  [{:keys [details cache_field_values_schedule metadata_sync_schedule] :as _database}]
-  (when-not (:let-user-control-scheduling details)
-    (let [random-defaults (sync.schedules/schedule-map->cron-strings (sync.schedules/default-schedule))
-          old-default-keys (cond-> []
-                             (contains? sync.schedules/default-cache-field-values-schedule-cron-strings cache_field_values_schedule)
-                             (conj :cache_field_values_schedule)
-
-                             (contains? sync.schedules/default-metadata-sync-schedule-cron-strings metadata_sync_schedule)
-                             (conj :metadata_sync_schedule))]
-      (not-empty (select-keys random-defaults old-default-keys)))))
```

Comment:
Just to check, we're still randomizing the schedules when we create a new Database, right? IIRC you added that in #14343 so we don't blast people with a ton of simultaneous syncs or something like that

Reply:
Should be. We are still randomizing them on startup for any that are default in this pr:
```clojure
(defmethod task/init! ::SyncDatabases
  [_]
  (job-init)
  ;; prevent "stampedes" on databases which all used the same sync schedules. Randomize them
  (let [dbs-with-default-schedules (->> (db/select Database)
                                        (filter
                                         (every-pred
                                          metabase-controls-schedule? old-default-schedule?)))]
    (when (seq dbs-with-default-schedules)
      (log/info (trs "Updating databases without custom sync scheduling to random schedules.")))
    (doseq [db dbs-with-default-schedules]
      (db/update! Database (u/the-id db)
        (sync.schedules/schedule-map->cron-strings (sync.schedules/default-schedule))))))
```

So we still have this code present which runs on startup to essentially migrate preexisting dbs to use a randomized schedule. I'm gonna track down the code that ensures dbs start with a randomized schedule as well.

Reply:
and yes new databases are given random schedules on line 598 of api/database: https://github.com/metabase/metabase/blob/master/src/metabase/api/database.clj#L598

Reply:
Nvm I see the code wherer that's happening below